### PR TITLE
add the crypto_kx api

### PR DESCRIPTION
**Pull request by mafintosh**
needs tests. /cc @emilbayes

### CODE REVIEW

**Comment by creationix** [comment]
I'm having trouble testing this.  

```js
const {
  crypto_kx_keypair,
  crypto_kx_client_session_keys,
  crypto_kx_server_session_keys,
  crypto_kx_PUBLICKEYBYTES,
  crypto_kx_SECRETKEYBYTES,
  crypto_kx_SESSIONKEYBYTES
} = require('./sodium-native')

// Allocate keys
let client = {
  pk: Buffer.alloc(crypto_kx_PUBLICKEYBYTES),
  sk: Buffer.alloc(crypto_kx_SECRETKEYBYTES),
  tx: Buffer.alloc(crypto_kx_SESSIONKEYBYTES),
  rx: Buffer.alloc(crypto_kx_SESSIONKEYBYTES)
}

let server = {
  pk: Buffer.alloc(crypto_kx_PUBLICKEYBYTES),
  sk: Buffer.alloc(crypto_kx_SECRETKEYBYTES),
  tx: Buffer.alloc(crypto_kx_SESSIONKEYBYTES),
  rx: Buffer.alloc(crypto_kx_SESSIONKEYBYTES)
}

// Generate keypairs
crypto_kx_keypair(server.pk, server.sk)
crypto_kx_keypair(client.pk, client.sk)

// Client calculates shared keys
crypto_kx_client_session_keys(
  client.rx, client.tx, client.pk, client.sk, server.pk)
// Server calculates shared keys
crypto_kx_server_session_keys(
  server.rx, server.tx, server.pk, server.sk, client.pk)

console.log({server, client})

console.log(server.rx.toString('hex'))
console.log(client.tx.toString('hex'))
console.log(server.tx.toString('hex'))
console.log(client.rx.toString('hex'))
```

I'm getting 4 different keys at the end, it should be two matching pairs.

File: binding.cc [hunk 1]
```diff
@@ -100,6 +100,43 @@ NAN_METHOD(sodium_compare) {
...
+}
+
+NAN_METHOD(crypto_kx_server_session_keys) {
+  ASSERT_BUFFER_MIN_LENGTH(info[0], rx, crypto_kx_SESSIONKEYBYTES)
+  ASSERT_BUFFER_MIN_LENGTH(info[1], tx, crypto_kx_SESSIONKEYBYTES)
+  ASSERT_BUFFER_MIN_LENGTH(info[2], server_pk, crypto_kx_PUBLICKEYBYTES)
+  ASSERT_BUFFER_MIN_LENGTH(info[3], server_sk, crypto_kx_SECRETKEYBYTES)
+  ASSERT_BUFFER_MIN_LENGTH(info[4], client_pk, crypto_kx_PUBLICKEYBYTES)
+
+  CALL_SODIUM(crypto_kx_client_session_keys(CDATA(rx), CDATA(tx), CDATA(server_pk), CDATA(server_sk), CDATA(client_pk)))
```
Comment (creationix) [inline binding.cc]: I think you meant `crypto_kx_server_session_keys` here.

Comment (creationix) [inline binding.cc]: I think you meant `crypto_kx_server_session_keys` here.

**Comment by creationix** [comment]
Found a typo in the bindings.  It works after it's fixed.

**Comment by mafintosh** [comment]
Added tests so @emilbayes is happy. Will merge after CI is green

**Comment by emilbayes** [comment]
Could you test the constants too, like we do with pwhash? Also, should the test be in sodium-test too?

**Comment by mafintosh** [comment]
@emilbayes added tests for constants. will move to sodium test when i port the api to js

**Pull request merged**